### ISSUE TITLE
Overflow or underflow happening for simple addition?

### ISSUE THREAD

**Issue by pascalriquier** [unitsofmeasurement/indriya#421]
We want to do a simple addition of 2 quantities, which are in unit (kg/(ha.year.year)), see code example below.
This is not giving us the correct answer, we assume due to the fact that the two numbers are converted to the system-unit, which will convert the year to seconds and then to the second power of that. Could this be handled better or could we do anything else on our side?

```
package be.vlaanderen.omgeving.pasberekening.domain.trend;


import static org.junit.jupiter.api.Assertions.assertEquals;
import static tech.units.indriya.quantity.Quantities.getQuantity;

import javax.measure.Quantity;
import javax.measure.Unit;
import org.junit.jupiter.api.Test;
import si.uom.NonSI;
import tech.units.indriya.unit.Units;

class QuantitiesTest {
  public static interface DepositionChange extends Quantity<DepositionChange> {
  }

  public static final Unit<DepositionChange> KILOGRAM_PER_HECTARE_PER_YEAR_PER_YEAR =
      Units.KILOGRAM.divide(NonSI.HECTARE.multiply(NonSI.YEAR_CALENDAR.pow(2)))
          .asType(DepositionChange.class);

  @Test
  void test() {
    assertEquals(
        getQuantity(4, KILOGRAM_PER_HECTARE_PER_YEAR_PER_YEAR)
            .add(getQuantity(2.25, KILOGRAM_PER_HECTARE_PER_YEAR_PER_YEAR)),
        getQuantity(6.25, KILOGRAM_PER_HECTARE_PER_YEAR_PER_YEAR));
  }


}

```

**Comment by pascalriquier** [comment]
Excuse, me I should have been more precise about that, the actual value is 
```
-5.342751485506088923587863698926158782093908691025920000 kg/(ha·day*365²)
````

It is not a rounding issue.

**Comment by andi-huber** [comment]
Agreed that's definitely not what we'd expect.

**Comment by pascalriquier** [comment]
@andi-huber for further clarity, we are using version 2.1.3 of indriya, do you want me to verify if the issue is also there in 2.2?

**Comment by andi-huber** [comment]
@pascalriquier yes that would be great thanks!

**Comment by pascalriquier** [comment]
OK @andi-huber I see that it is fixed in 2.2, we know what to do ;-)

**Comment by andi-huber** [comment]
No longer an issue in `2.2`.
